### PR TITLE
v7 - upgrade to angular 5 compiler 

### PR DESCRIPTION
**Pull request by e-schultz**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,35 +53,35 @@
...
     "core-js": "^2.4.1",
     "jasmine": "^2.5.3",
     "nodemon": "^1.11.0",
     "nyc": "^10.2.0",
-    "redux": "^3.5.0",
-    "reflect-metadata": "0.1.3",
+    "redux": "^3.7.2",
+    "reflect-metadata": "0.1.10",
     "rimraf": "^2.5.2",
     "rxjs": "^5.0.1",
```
Comment (leo6104) [inline package.json]: Please update rxjs to ^5.5.2

File: src/components/root-store.ts [hunk 2]
```diff
@@ -20,7 +20,12 @@ import 'rxjs/add/operator/filter';
 import 'rxjs/add/operator/distinctUntilChanged';
```
Comment (leo6104) [inline src/components/root-store.ts]: use lettable operators for reducing bundle size. 
```
import 'rxjs/operators/distinctUntilChanged';
import 'rxjs/operators/switchMap';
```

File: src/components/root-store.ts [hunk 3]
```diff
@@ -36,22 +41,26 @@ export class RootStore<RootState> extends NgRedux<RootState> {
     NgRedux.instance = this;
     this._store$ = new BehaviorSubject<RootState | undefined>(undefined)
```
Comment (leo6104) [inline src/components/root-store.ts]: use pipe for lettable operators.

File: src/components/root-store.ts [hunk 4]
```diff
@@ -64,34 +73,37 @@ export class RootStore<RootState> extends NgRedux<RootState> {
...
 
-  select = <S>(
-    selector?: Selector<RootState, S>,
-    comparator?: Comparator): Observable<S> =>
+  select = <SelectedType>(
+    selector?: Selector<RootState, SelectedType>,
+    comparator?: Comparator
+  ): Observable<SelectedType> =>
     this._store$
       .distinctUntilChanged()
```
Comment (leo6104) [inline src/components/root-store.ts]: use pipe for lettable operators.

File: src/components/sub-store.spec.ts [hunk 5]
```diff
@@ -28,72 +30,75 @@ describe('Substore', () => {
...
   it('handles path selection on a base path that doesn\'t exist yet', () => {
     const nonExistentSubStore = ngRedux.configureSubStore(
       ['sure', 'whatever'],
-      (state: any, action: any) => ({ ...state, value: action.newValue }));
-    nonExistentSubStore.select([ 'value' ])
+      (state: any, action: any) => ({ ...state, value: action.newValue })
+    );
+    nonExistentSubStore
+      .select(['value'])
       .take(2)
```
Comment (leo6104) [inline src/components/sub-store.spec.ts]: use pipe for lettable operators.

File: src/components/sub-store.ts [hunk 6]
```diff
@@ -8,49 +8,58 @@ import {
...
+      localReducer
+    );
 
   select = <SelectedState>(
     selector?: Selector<State, SelectedState>,
-    comparator?: Comparator): Observable<SelectedState> => this.rootStore
+    comparator?: Comparator
+  ): Observable<SelectedState> =>
+    this.rootStore
       .select(this.basePath)
```
Comment (leo6104) [inline src/components/sub-store.ts]: use pipe for lettable operators.

**Comment by e-schultz** [comment]
@leo6104 good idea - going to do that. Thanks.

**Comment by tigredonorte** [comment]
waiting for angular5 support

**Comment by e-schultz** [comment]
@tigredonorte v6.6.0-1 should support Angular 5, although I don't have any large applications upgraded yet to make use of it. It still uses the v4 compiler/etc under the hood - but works with v5

This current branch is published under v7.0.0-3 - which uses the updated compiler (although not backwards compatible with v4), and includes changes like using lettable operators. 

I'm a little pressed for time, but if anyone is able to give it a test - would be great. There was nothing specifically breaking in v4 to v5 - and was mostly just a matter of updating peer dependencies in the v6 branch.

**Comment by murraybauer** [comment]
I upgraded an existing app to angular 5, changed rxjs syntax and upgraded to angular-store v7.0.0-3.  No adverse effects that I can see.

**Pull request merged**